OpenRA.Mods.D2k/Traits/Render/WithCrumbleOverlay.cs

```diff
@@ -35,21 +35,32 @@ public class WithCrumbleOverlayInfo : ConditionalTraitInfo, Requires<RenderSprit
 
 	public class WithCrumbleOverlay : ConditionalTrait<WithCrumbleOverlayInfo>
 	{
+		readonly Animation overlay;
+		readonly RenderSprites renderSprites;
+		readonly WithCrumbleOverlayInfo info;
+		readonly AnimationWithOffset anim;
+
 		public WithCrumbleOverlay(ActorInitializer init, WithCrumbleOverlayInfo info)
 			: base(info)
 		{
+			this.info = info;
+
 			if (init.Contains<SkipMakeAnimsInit>())
 				return;
 
-			var rs = init.Self.Trait<RenderSprites>();
+			renderSprites = init.Self.Trait<RenderSprites>();
 
-			var overlay = new Animation(init.World, rs.GetImage(init.Self));
-			var anim = new AnimationWithOffset(overlay, null, () => IsTraitDisabled);
+			overlay = new Animation(init.World, renderSprites.GetImage(init.Self));
+			anim = new AnimationWithOffset(overlay, null, () => IsTraitDisabled);
 
-			// Remove the animation once it is complete
-			overlay.PlayThen(info.Sequence, () => init.World.AddFrameEndTask(w => rs.Remove(anim)));
+			renderSprites.Add(anim, info.Palette, info.IsPlayerPalette);
```

Comment:
Won't this cause the animation to be visible but stuck on the first frame if the trait stays disabled?
I would expect this to be added when starting the animation below.

Reply:
This will crash if you put the trait on vehicles, so I will safely move it.